qemu/tests/usbdev.py

```diff
@@ -0,0 +1,428 @@
+        status, output = self.session.cmd_status_output(mount_cmd, timeout=self.cmd_timeout)
+        if status != 0:
+            logging.debug("The devices which will be mounted are: %s" % devname )
+            raise error.TestFail("Failed to mount with error: %s" % output)
+
+    def umount_usb_disk(self, devname):
+        
+        """
+        unmount usb disk
+        """ 
+
+        umount_cmd = self.umount_cmd % devname
+        if umount_cmd:
+            error.context("Unmounting usb disk with cmd umount", logging.info)
+        status, output = self.session.cmd_status_output(umount_cmd, timeout=self.cmd_timeout)
+        if status != 0:
+            logging.debug("The mounted devices are: %s" % devname)
+            raise error.TestFail("Failed to umount with error: %s" % output)
+
+    def check_mounting(self, devname):
+ 
+        """
+        check if usb disk was mounted
+        """
+        error.context("Check if the usb disk was mounted" , logging.info)
+        status, output = self.session.cmd_status_output(self.check_mount_cmd, timeout=self.cmd_timeout)
+        if status != 0:
+            logging.debug("Can't run the command of mount " )
+            raise error.TestFail("Failed to run mount conmmand with error: %s" % output)
+        if devname in output:
+            return True
+        else:
+            return False
+ 
+    def do_io_test_guest(self,devname):
+      
+        """
+        do io test on usb disk with fs
+        """
+
+        if not self.check_mounting(devname):
+            raise error.TestFail("The usb disk wasn't mounted")
+        testfile_name =self.testfile_name
+        if testfile_name:
+            error.context("Write some random string to test file", logging.info)
+            ranstr = utils_misc.generate_random_string(100)
+
+            writefile_cmd = self.writefile_cmd % (ranstr, testfile_name)
+            status, output = self.session.cmd_status_output(writefile_cmd, timeout=self.cmd_timeout)
+            if status != 0:
+                raise error.TestFail("Write to file error: %s" % output)
+
+            error.context("Read in the file to see whether content has changed",
+                      logging.info)
+            if self.md5chk_cmd:
+                status, output = self.session.cmd_status_output(self.md5chk_cmd, timeout=self.cmd_timeout)
+                if status != 0:
+                    raise error.TestFail("Check file md5sum error.")
+
+            readfile_cmd = self.readfile_cmd % testfile_name
+            status, output = self.session.cmd_status_output(readfile_cmd, timeout=self.cmd_timeout)
+            if status != 0:
+                raise error.TestFail("Read file error: %s" % output)
+            if output.strip() != ranstr:
+                raise error.TestFail("The content written to file has changed, "
+                                 "from: %s, to: %s" % (ranstr, output.strip()))
+    
+    def create_partition(self):
```

Comment:
hi @xutian the function format_guest_disk in utils_misc can't be used if the disk didn't create a new disklabel of label-type,and the function contain three step: create partition, format partition , mount partition,I think dispart them will be better ,so I think i should keep my mind about this case.